cmd/createorupdate/update.go

```diff
@@ -43,6 +52,42 @@ func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin)
 	return nil
 }
 
+// needUpdate returns a map with the exact operations that need to be
+// executed during an upgrade. Only scale is supported for now.
+func needUpdate(cs, oldCs *api.OpenShiftManagedCluster) map[string]bool {
```

Comment:
Eventually we will need to pull in here azuredeploy.json (both old and new) in order to figure out whether the CSE has changed.